lib/maths/CBoostedTree.cc

```diff
@@ -316,6 +316,26 @@ class CLeafNodeStatistics final {
     //! Get the row mask for this leaf node.
     core::CPackedBitVector& rowMask() { return m_RowMask; }
 
+    //! Estimate maximum leaf statistics bookkeeping memory for training
+    //! boosted trees on data frames with \p numberRows rows, \p numberCols columns
+    //! with specified settings for \p featureBagFraction and \p numberSplitsPerFeature
+    static size_t estimateMemoryUsage(std::size_t numberRows,
+                                      std::size_t numberCols,
+                                      double featureBagFraction,
+                                      std::size_t numberSplitsPerFeature) {
+        std::size_t featureBagSize{
+            static_cast<std::size_t>(ceil(featureBagFraction) * (numberCols - 1)) *
+            sizeof(std::size_t)};
+        std::size_t rowMaskSize{numberRows / 32};
+        std::size_t gradientsSize{(numberCols - 1) * numberSplitsPerFeature *
+                                  sizeof(CFloatStorage)};
+        std::size_t curvatureSize{gradientsSize};
+        std::size_t missingGradientsSize{(numberCols - 1) * sizeof(CFloatStorage)};
```

Comment:
These guys all use doubles, i.e. see [this](https://github.com/elastic/ml-cpp/blob/41846a6300d4011effff54b5a591bb1c4e07b7bd/lib/maths/CBoostedTree.cc#L525) line etc, so this and previous should be `sizeof(double)`.

Reply:
done